src/lib/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Factory/RandomSortClauseHandlerFactory.php

```diff
@@ -32,14 +32,14 @@ public function __construct(Connection $connection, iterable $randomSortClauseGa
      */
     public function getGateway(): AbstractRandom
     {
-        $driverName = $this->connection->getDatabasePlatform()->getName();
+        $databasePlatform = $this->connection->getDatabasePlatform();
 
         foreach ($this->randomSortClauseGateways as $gateway) {
-            if ($gateway->getDriverName() === $driverName) {
+            if ($gateway->supportsPlatform($databasePlatform)) {
                 return $gateway;
             }
         }
 
-        throw new InvalidArgumentException('$this->randomSortClauseGateways', 'No RandomSortClauseHandler found for driver ' . $driverName);
+        throw new InvalidArgumentException('$this->randomSortClauseGateways', 'No RandomSortClauseHandler found for driver ' . get_class($databasePlatform));
```

Comment:
Besides formatting, please use `sprintf` here.
```suggestion
        throw new InvalidArgumentException(
            '$this->randomSortClauseGateways',
            'No RandomSortClauseHandler found for driver ' . get_class($databasePlatform)
        );
```